### PR TITLE
Posts Soft Delete

### PR DESCRIPTION
**Pull request by monocliffe**
[TPOZ-1471](https://wpengine.atlassian.net/browse/TPOZ-1471)

Implemented soft delete functionality using the Discard gem. Added soft delete migration. Added delete buttons to single post view and index view. Implemented restore method that undos soft delete. Restore available while delete successful flash banner exists.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -26,12 +27,26 @@ def edit; end
...
+  def destroy
+    @post.discard
+    flash[:alert] = "Post Removed. #{view_context.link_to('Undo', restore_post_path(@post))}".html_safe
+    redirect_to posts_url
+  end
+
+  def restore
+    @post.undiscard
+    flash[:success] = 'Post recovered successfully!'
+    redirect_to posts_url
```
Comment (JamesDominy) [inline app/controllers/posts_controller.rb]: If you've just restored a post, you probably want to be redirected to that post directly.

**Review by JamesDominy — COMMENTED** [review]
Looks good. Do you have a way to control how long the deleted posts stick around in the database? Ideally they should have an expiry timestamp, after which they are hard deleted.

**Comment by monocliffe** [comment]
> Looks good. Do you have a way to control how long the deleted posts stick around in the database? Ideally they should have an expiry timestamp, after which they are hard deleted.

Currently no, but it shouldn't be too much work to implement a cron job that deletes posts over a certain age.

**Review by tmwpengine — APPROVED** [review]
LGTM!

**Pull request merged**